File: vijos/CPP/vijos_1131.cpp
//
// Created by 弱冠 on 2018/7/19.
//
#include <cmath>
#include <cstdio>
typedef struct node {
    int x0;
    int y0;
}Node;

Node is_max_min(int P, int Q) {
    int m = P;
    int n = Q;
    while (m != n) {
        if (m > n) {
            m = m -n;
        } else if (n > m) {
            n = n - m;
        }
    }
    Node node1;
    node1.x0 = m;
    node1.y0 = P * Q /m;
    return node1;
}

int main() {
    int x, y;
    scanf("%d %d", &x, &y);
    int sqrt_xy = (int)sqrt(x*y);
    int count = 0;
    int n = 0;
    int i = x * y;
    while (i >= sqrt_xy) {
        int j = sqrt_xy;
        while (j > 0) {
            if (i * j == x * y) {
                Node node1 = is_max_min(i, j);
                if (node1.x0 == x && node1.y0 == y) {
                    if (i == sqrt_xy && j == sqrt_xy) {
                        n = 1;
                    }
                    count++;
                }
            }
            j -= 1;
        }
        i -= 1;
    }
    if (n) {
        printf("%d\n", count*2 -1);
    } else {
        printf("%d\n", count*2);
    }
    return 0;
}

File: vijos/Python/vijos_1131.py
import math

x, y = (int(x) for x in input().strip().split(" "))

i = x * y

def is_max_min(P, Q):
    m = P
    n = Q
    while m != n:
        if m > n:
            m = m - n
        elif n > m:
            n = n -m
    x0 = m
    y0 = P * Q / x0
    return x0, y0


count = 0
n = False

sqrt_xy = int(math.sqrt(x*y))
while i >= sqrt_xy:
    j = sqrt_xy
    while j > 0:
        if i * j == x * y:
            x0, y0 = is_max_min(i, j)
            if x0 == x and y0 == y:
                if i == sqrt_xy and j == sqrt_xy:
                   n = True
                count += 1
        j -= 1
    i -= 1


if n:
    print(count * 2 -1)
else:
    print(count * 2)

